In [5]:
import numpy as np
import scipy.stats as stats

ЗАДАЧА №1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания А с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [6]:
# Из условия задачи знаем:
σ = 16
μ = 80
n = 256

In [7]:
# Сигма генеральной совокупности известна, поэтому используем z-критерий:
Z = stats.norm.ppf(0.975)
Z

1.959963984540054

In [8]:
# Определяем интервалы:
z_left = μ - Z * (σ/np.sqrt(n))
z_right = μ + Z * (σ/np.sqrt(n))
z_left, z_right

(78.04003601545995, 81.95996398454005)

ОТВЕТ ЗАДАЧИ №1: Доверительный интервал для оценки будет от 78,04 до 81,96.

ЗАДАЧА №2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1. Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.


In [9]:
# Из условия задачи знаем:
n = 10
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [10]:
# Посчитаем среднее значение и сигму выборки:
μ = np.mean(x)
σ = np.std(x,ddof=1)
μ, σ

(6.590000000000001, 0.4508017549014448)

In [11]:
# Сигма генеральной совокупности неизвестна, будем использовать t-критерий Стьюдента для расчета:
t = stats.t.ppf(0.975, (n - 1))
t

2.2621571627409915

In [12]:
# Определяем интервалы:
z_left = μ - t * (σ/np.sqrt(n))
z_right = μ + t * (σ/np.sqrt(n))
z_left, z_right

(6.267515851415713, 6.912484148584288)

ОТВЕТ ЗАДАЧИ №2: Доверительный интервал будет от 6,28 до 6,90.

ЗАДАЧА №3. Рост дочерей (детей): 175, 167, 154, 174, 178, 148, 160, 167, 169, 170.
Рост матерей (родителей):  178, 165, 165, 173, 168, 155, 160, 164, 178, 175
Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

In [13]:
x = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
y = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])

In [14]:
# Посчитаем средние значение по выборкам и значение разности:
μ_x = np.mean(x)
μ_y = np.mean(y)
Δ = round((μ_x - μ_y), 2)
μ_x, μ_y, Δ

(166.2, 168.1, -1.9)

In [15]:
# Посчитаем значения сигм по обеим выборкам:
σ_x = np.std(x)
σ_y = np.std(y)
σ_x, σ_y

(9.05317623820502, 7.354590403278759)

In [16]:
# Общую дисперсию рассчитываем по формуле:
D = (σ_x**2 + σ_y**2)/2
D

68.025

In [17]:
# Теперь можно посчитать стандартную ошибку среднего:
S = np.sqrt(D * (1/len(x) + 1/len(y)))
S

3.6884956283015984

In [18]:
# Сигма генеральной совокупности неизвестна, будем использовать t-критерий Стьюдента для расчета:
t = stats.t.ppf(0.975, (len(x) + len(y) - 2))
t

2.10092204024096

In [19]:
# Посчитаем интервалы:
z_left = Δ - t * S
z_right = Δ + t * S
z_left, z_right

(-9.649241760831256, 5.849241760831257)

ОТВЕТ ЗАДАЧИ №3: 95%-й интервал для разности роста: от -9.65 до 5.85.